statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1215 @@
+
+        Returns
+        -------
+        stat : float
+            The chi^2 test statistic.
+        df : integer
+            The degrees of freedom of the reference distribution
+        pvalue : float
+            The p-value for the test.
+        """
+        stat = np.asarray(self.chi2_contribs).sum()
+        df = np.prod(np.asarray(self.table.shape) - 1)
+        pvalue = 1 - stats.chi2.cdf(stat, df)
+        b = _bunch()
+        b.stat = stat
+        b.df = df
+        b.pvalue = pvalue
+        return b
+
+
+    def ordinal_association(self, row_scores=None, col_scores=None):
+        """
+        Assess independence between rows and columns for ordinal factors.
+
+        This is the 'linear by linear' association test.
+
+        Parameters
+        ----------
+        row_scores : array-like
+            An array of numeric row scores
+        col_scores : array-like
+            An array of numeric column scores
+
+        Returns a bunch with the following attributes:
+
+        stat : float
+            The test statistic.
+        stat_e0 : float
+            The expected value of the test statistic under the null
+            hypothesis.
+        stat_sd0 : float
+            The standard deviation of the test statistic under the
+            null hypothesis.
+        zscore : float
+            The Z-score for the test statistic.
+        pvalue : float
+            The p-value for the test.
+
+        Notes
+        -----
+        Using the default row and column scores for the linear-by-linear
+        association test gives the Cochran-Armitage trend test.
+        """
+
+        if row_scores is None:
+            row_scores = np.arange(self.table.shape[0])
+
+        if col_scores is None:
+            col_scores = np.arange(self.table.shape[1])
+
+        if len(row_scores) != self.table.shape[0]:
+            raise ValueError("The length of `row_scores` must match the first dimension of `table`.")
+
+        if len(col_scores) != self.table.shape[1]:
+            raise ValueError("The length of `col_scores` must match the second dimension of `table`.")
+
+        # The test statistic
+        stat = np.dot(row_scores, np.dot(self.table, col_scores))
+
+        # Some needed quantities
+        n_obs = self.table.sum()
+        rtot = self.table.sum(1)
+        um = np.dot(row_scores, rtot)
+        u2m = np.dot(row_scores**2, rtot)
+        ctot = self.table.sum(0)
+        vn = np.dot(col_scores, ctot)
+        v2n = np.dot(col_scores**2, ctot)
+
+        # The null mean and variance of the test statistic
+        e_stat = um * vn / n_obs
+        v_stat = (u2m - um**2 / n_obs) * (v2n - vn**2 / n_obs) / (n_obs - 1)
+        sd_stat = np.sqrt(v_stat)
+
+        zscore = (stat - e_stat) / sd_stat
+        pvalue = 2 * stats.norm.cdf(-np.abs(zscore))
+
+        b = _bunch()
+        b.stat = stat
+        b.stat_e0 = e_stat
+        b.stat_sd0 = sd_stat
+        b.zscore = zscore
```

Comment:
terminology: zscore is specific to normal.

We don't have a fully establishing naming convention for this yet (scipy.stats is also still inconsistent in what the returned `statistic` is, AFAIK)

I'm trying to move towards, `effect`, `effect_size` (standardized effect) and `statistic` for the actual test statistic (i.e. zscore here. But I'm not sure yet about all the different versions of "statistics".